privacyidea/api/lib/utils.py

```diff
@@ -247,19 +247,44 @@ def verify_auth_token(auth_token, required_role=None):
     :param required_role: list of "user" and "admin"
     :return: dict with authtype, realm, rights, role, username, exp, nonce
     """
+    r = None
     if required_role is None:
         required_role = ["admin", "user"]
     if auth_token is None:
         raise AuthError(_("Authentication failure. Missing Authorization header."),
                         id=ERROR.AUTHENTICATE_AUTH_HEADER)
-    try:
-        r = jwt.decode(auth_token, current_app.secret_key, algorithms=['HS256'])
-    except jwt.DecodeError as err:
-        raise AuthError(_("Authentication failure. Error during decoding your token: {0!s}").format(err),
-                        id=ERROR.AUTHENTICATE_DECODING_ERROR)
-    except jwt.ExpiredSignature as err:
-        raise AuthError(_("Authentication failure. Your token has expired: {0!s}").format(err),
-                        id=ERROR.AUTHENTICATE_TOKEN_EXPIRED)
+
+    headers = jwt.get_unverified_header(auth_token)
+    if headers.get("alg") == "RS256":
+        # The trusted JWTs are RSA signed
+        trusted_jwts = current_app.config.get("PI_TRUSTED_JWT", [])
+        for trusted_jwt in trusted_jwts:
+            try:
+                if trusted_jwt.get("algorithm") == "RS256":
+                    j = jwt.decode(auth_token,
+                                   trusted_jwt.get("public_key"),
+                                   algorithms=["RS256"])
```

Comment:
I think we should not limit the algorithms to 'RS256'.
With ``jwt.decode(auth_token, public_key, algorithms=list(jwt.algorithms.requires_cryptography))`` we can enforce to use only public-key crypto algorithms for the token verification.
Or we could use a list of allowed algorithms (to also include HMAC with a shared secret).

Reply:
Also we should at least use RSA-PSS (algorithms ``PS256``, ``PS384`` and ``PS512``) since PKCS1v15 is deprecated.

Reply:
You are right. Will do.

Reply:
done in  #a3e20f13